python/lang/correctness/is-function-without-parentheses.yaml

```diff
@@ -0,0 +1,17 @@
+rules:
+- id: is-function-without-parentheses
+  languages:
+  - python
+  message: Is "$FUNC" a function or an attribute? If it is a function, you may have
+    meant $X.$FUNC() because $X.$FUNC is always true.
+  patterns:
+  - pattern: $X.$FUNC
+  - pattern-not-inside: $X.$FUNC()
+  - metavariable-regex:
+      metavariable: $FUNC
+      regex: is_.*
+  severity: WARNING
+  metadata:
+    category: correctness
```

Comment:
I think this rule may be better qualified as `maintainability` - @chmccreery what do you think?

Reply:
Sure! I don't really understand the categories, tbh, I just put most things that I see as likely bugs (rather than security-specific) as correctness. I trust you more than me here 👍

Reply:
They're fuzzy categories anyway 😝 

My usual logic is, "will this crash the program?" If so, it's probably correctness. If not, it's probably best-practice or maintainability 🤷

Reply:
Oh, well the eqeq rule is labeled as "correctness" and that was my benchmark - and definitely will not crash any programs 😄

Reply:
maybe "unexpected-behavior" is the category I'm looking for